src/plugins/unified_search/public/search_bar/search_bar.tsx

```diff
@@ -42,572 +42,426 @@ export interface SearchBarInjectedDeps {
-  displayStyle?: 'inPage' | 'detached';
-  // super update button background fill control
-  fillSubmitButton?: boolean;
-  dataViewPickerComponentProps?: DataViewPickerProps;
-  textBasedLanguageModeErrors?: Error[];
   onTextBasedSavedAndExit?: ({ onSave }: OnSaveTextLanguageQueryProps) => void;
-  showSubmitButton?: boolean;
-  submitButtonStyle?: QueryBarTopRowProps['submitButtonStyle'];
-  // defines size of suggestions query popover
-  suggestionsSize?: SuggestionsListSize;
-  isScreenshotMode?: boolean;
-
-  /**
-   * Disables all inputs and interactive elements,
-   */
-  isDisabled?: boolean;
 }
 
 export type SearchBarProps<QT extends Query | AggregateQuery = Query> = SearchBarOwnProps<QT> &
   SearchBarInjectedDeps;
 
-interface State<QT extends Query | AggregateQuery = Query> {
-  isFiltersVisible: boolean;
-  openQueryBarMenu: boolean;
-  showSavedQueryPopover: boolean;
-  currentProps?: SearchBarProps;
-  query?: QT | Query;
-  dateRangeFrom: string;
-  dateRangeTo: string;
-}
-
-class SearchBarUI<QT extends (Query | AggregateQuery) | Query = Query> extends Component<
-  SearchBarProps<QT> & WithEuiThemeProps,
-  State<QT | Query>
-> {
-  public static defaultProps = {
-    showQueryMenu: true,
-    showFilterBar: true,
-    showDatePicker: true,
-    showSubmitButton: true,
-    showAutoRefreshOnly: false,
-  };
-
-  private services = this.props.kibana.services;
-  private savedQueryService = this.services.data.query.savedQueries;
-
-  public static getDerivedStateFromProps(
-    nextProps: SearchBarProps,
-    prevState: State<AggregateQuery | Query>
-  ) {
-    if (isEqual(prevState.currentProps, nextProps)) {
-      return null;
-    }
-
-    let nextQuery = null;
-    if (
-      nextProps.query &&
-      isOfQueryType(nextProps.query) &&
-      nextProps.query.query !== get(prevState, 'currentProps.query.query')
-    ) {
-      nextQuery = {
-        query: nextProps.query.query,
-        language: nextProps.query.language,
-      };
-    } else if (
-      nextProps.query &&
-      prevState.query &&
-      isOfQueryType(nextProps.query) &&
-      isOfQueryType(prevState.query) &&
-      nextProps.query.language !== prevState.query.language
-    ) {
-      nextQuery = {
-        query: '',
-        language: nextProps.query.language,
-      };
-    } else if (nextProps.query && !isOfQueryType(nextProps.query)) {
-      nextQuery = nextProps.query;
+function SearchBarUI<QT extends (Query | AggregateQuery) | Query = Query>({
+  customSubmitButton,
+  dateRangeFrom,
+  dateRangeTo,
+  dataTestSubj,
+  dataViewPickerComponentProps,
+  displayStyle,
+  fillSubmitButton,
+  filters,
+  hiddenFilterPanelOptions,
+  iconType,
+  indexPatterns,
+  indicateNoData,
+  isClearable,
+  isDisabled,
+  isLoading,
+  isRefreshPaused,
+  isScreenshotMode,
+  nonKqlMode,
+  placeholder,
+  query,
+  refreshInterval,
+  savedQuery,
+  screenTitle,
+  showAutoRefreshOnly = false,
```

Comment:
💬 Previous implementation used defaultProps for some props. Included these here.